.gitignore

```diff
@@ -31,6 +31,12 @@
 ipch/
 *.VC.db
 
+# Visual Studio Code
+.vscode/*
+!.vscode/settings.json

```

Comment:
I mis-read this earlier, let's not be including these in the tree as the VSCode system is still in flux and not officially supported